### PR TITLE
Roll up all the PRs into one : )

### PR DESCRIPTION
**Pull request by rreusser**
I know I've overwhelmed you with PRs. This PR rolls up all of the other PRs so that you don't have to worry about the not-100%-trivial merging. I'm glad to make any requested changes or pull out parts if they don't fit, but thought this might save a bit of mental overhead on your part. If this works for you, then can ignore the other PRs. 👍 

Thanks!!

### CODE REVIEW

File: components/styles/interval.css [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  -ms-user-select: none;
+  user-select: none;
+}
+.control-panel-interval-{{ UUID }} .control-panel-interval-handle {
+  background-color: {{ INTERVAL_COLOR }};
+  position: absolute;
+  height: 20px;
+  min-width: 1px;
+}
+#control-panel{{ UUID }}.control-panel-interval-dragging * {
```
Comment (freeman-lab) [inline components/styles/interval.css]: should there be a hyphen here for consistency with the others e.g. `#control-panel-{{ UUID }}`? might need another change elsewhere

Comment (rreusser) [inline components/styles/interval.css]: Ah, good catch! I know this is all a lot of code. Semi-sorry about that.

**Comment by freeman-lab** [comment]
This is fantastic @rreusser! I left one note that might be worth looking at, but otherwise think it all looks good to go. Will be really fun to have these!

File: components/styles/interval.css (hunk 1 above)
Comment (freeman-lab) [inline components/styles/interval.css]: No don't be, it's awesome!

**Comment by rreusser** [comment]
Small notes:
- Give it a once-over for `semistandard` formatting
- Figure out how to get the tests running. They're minimal but at least rule out syntax errors.
- Interval should probably really just accept a mapping function, with maybe a couple presets like 'linear' and 'log'. Relatively minimal changes, but would clean things up a bit.

**Comment by freeman-lab** [comment]
Ah, are you having trouble with `testron`? I've had some issues as well, mainly due to `electron` versioning I think. I'm opening + merging a small PR now that switches to `smokestack`, which I'm finding works a little more reliably. If you merge that in here you should be able to get the tests running, happy to help if there's trouble.

**Comment by freeman-lab** [comment]
Oh and re: `standard` I'm happy merging and then fixing that later.

**Comment by rreusser** [comment]
Okay, the additional changes made:

- The tests weren't actually testing anything except non-exceptions, I think. `typof document.querySelector('not-present') === 'object'` so that all tests succeeded even if the elements weren't present. I cleaned this up just a bit.
- Some components had a UUID attached to the selector, some didn't. Not strictly necessary if no custom styles, but for consistency, I added a UUID across the board.
- `standard` 👍
- a couple little style nits
- select menu merged into this PR

Lots more functionality that could be added, but good to go from my opinion :)

**Comment by freeman-lab** [comment]
looks fantastic 👍  merging in, thanks for these awesome additions !

**Comment by freeman-lab** [comment]
fyi just published this as 1.2.0

**Pull request merged**